src/IO/H5/CombineH5.cpp

```diff
@@ -373,35 +375,43 @@ void combine_h5_dat(const std::vector<std::string>& h5_files_to_combine,
                        << " doesn't match other H5 files.");
       }
 
-      // This is the first file. We can't make any decisions here so just store
-      // the number of times and the earliest time of this file
+      // This is the first (last) file. We can't make any decisions here so just
+      // store the number of times and the earliest time of this file
       if (not earliest_time.has_value()) {
         num_time_map.at(dat_filename)[index] = dimensions[0];
-        earliest_time = times(0, 0);
+        earliest_time = times[0][0];
         continue;
       }
 
-      // Determine if the earliest time of the previous file (previous = later
-      // in the sequence of files since we are looping backward) is before
-      // any of the times in this file. If so, don't include those times.
-      std::optional<size_t> row = times.rows() - 1;
-      while (times(row.value(), 0) >= earliest_time.value()) {
-        // Once we get to the first time in this file, if it's still after the
-        // earliest time in the previous file, then we don't include any times
-        // from this file.
-        if (row.value() == 0) {
-          row.reset();
-          break;
+      // Check that the earliest time of the previous file (previous = later
+      // in the sequence of files since we are looping backward) is after the
+      // first time of this file. We require the files to be passed in
+      // increasing order by their first time.
+      if (UNLIKELY(times[0][0] >= earliest_time.value())) {
+        ERROR_NO_TRACE("The H5 files passed in "
+                       << h5_files_to_combine
+                       << " are not monotonically increasing in their first "
+                          "times for dat file "
+                       << dat_filename << "");
+      }
+
+      // Determine if the earliest time of the previous file is before any of
+      // the times in this file. If so, don't include those times.
+      size_t row = times.size() - 1;
+      while (times[row][0] >= earliest_time.value()) {
+        // This should have been taken care of before so we should never get
```

Comment:
I had to think about this (terrible, I know), but maybe change to ~ `Make sure we don't reach row 0 since that would mean the files are not ordered. We should've checked for this above, so this is an additional safety check.`